File: api/api-facades/api-tests/src/main/java/org/eclipse/dirigible/api/v3/test/Process.java
/*
 * Copyright (c) 2017 SAP and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 * Contributors:
 * SAP - initial API and implementation
 */

package org.eclipse.dirigible.api.v3.test;

import java.util.ArrayList;
import java.util.List;

public class Process {

	private String name;

	private List<Task> tasks = new ArrayList<Task>();

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public List<Task> getTasks() {
		return tasks;
	}

	public Task createTask(String taskName) {
		Task task = new Task();
		task.setName(taskName);
		tasks.add(task);
		return task;
	}

	public boolean existsTask(Task task) {
		return tasks.contains(task);
	}

}

File: ide/ui/ide-monaco/src/main/resources/META-INF/dirigible/ide-monaco/monaco-editor/esm/vs/base/common/process.js
/*
 * Copyright (c) 2021 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
 * SPDX-License-Identifier: EPL-2.0
 */
import { isWindows, isMacintosh, setImmediate } from './platform.js';
var safeProcess = (typeof process === 'undefined') ? {
    cwd: function () { return '/'; },
    env: Object.create(null),
    get platform() { return isWindows ? 'win32' : isMacintosh ? 'darwin' : 'linux'; },
    nextTick: function (callback) { return setImmediate(callback); }
} : process;
export var cwd = safeProcess.cwd;
export var env = safeProcess.env;
export var platform = safeProcess.platform;
